src/bitwise.rs

```diff
@@ -0,0 +1,93 @@
+/// Contains some helper methods to manipulate bits,
+/// the index (`bit_idk`) is supposed to be from lsb to msb (right to left)
+pub(crate) struct Bitwise;
+
+impl Bitwise {
+    pub(crate) fn is_on(num: u32, bit_idx: u8) -> bool {
+        debug_assert!(bit_idx < 32);
+
+        let mask = 0b1 << bit_idx;
+        (num & mask) != 0
+    }
+
+    pub(crate) fn is_off(num: u32, bit_idx: u8) -> bool {
+        debug_assert!(bit_idx < 32);
+
+        let mask = 0b1 << bit_idx;
+        (num & mask) == 0
+    }
+
+    pub(crate) fn set_on(num: &mut u32, bit_idx: u8) {
+        debug_assert!(bit_idx < 32);
+
+        let mask = 0b1 << bit_idx;
+        *num |= mask;
+    }
+
+    pub(crate) fn set_off(num: &mut u32, bit_idx: u8) {
+        debug_assert!(bit_idx < 32);
+
+        let mask = !(0b1 << bit_idx);
+        *num &= mask;
+    }
+
+    /// Switches from 1 to 0, or conversely from 0 to 1
+    pub(crate) fn toggle_bit(num: &mut u32, bit_idx: u8) {
+        debug_assert!(bit_idx < 32);
+
+        let mask = 0b1 << bit_idx;
+        *num ^= mask;
+    }
+
+    pub(crate) fn set_bit(num: &mut u32, bit_idx: u8, value: u8) {
+        match value {
+            0 => Self::set_off(num, bit_idx),
+            1 => Self::set_on(num, bit_idx),
+            _ => panic!(),
+        }
+    }
+
+    pub(crate) fn get_bit(num: u32, bit_idx: u8) -> u8 {
+        debug_assert!(bit_idx < 32);
+
+        match Self::is_on(num, bit_idx) {
+            true => 1,
+            false => 0,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn basic_tests() {
```

Comment:
Here I prefer to split  the tests to recognize faster a future change that broke this implementation.  You can also take this test with all implementations, but I think  also it is better to have almost one test for every functionality.